react/src/screens/BookDetail/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component,Fragment } from 'react';
+import { NavLink } from 'react-router-dom';
+
+import BookDetailCommentaries from '../../components/BookDetailCommentaries';
+import BookDetailData from '../../components/BookDetailData';
+import BookDetailSuggestions from '../../components/BookDetailSuggestions';
+import {data} from '../../mock.js';
+import routes from '../../constants/routes';
+
+import './styles.css';
+
+class BookDetail extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: null,
+      errorInfo: null
+    };
+  }
+
+  componentDidCatch(error, errorInfo) {
+    // Catch errors in any child components and re-renders with an error message
+    this.setState({
+      error: error,
+      errorInfo: errorInfo
+    });
+  }
+  render() {
+
+    if (this.state.error) {
+      // Fallback UI if an error occurs
+      return (
+        <div>
```

Comment:
Use `Fragments` to wrap elements

Reply:
If you would like, we could extract this to a HOC
https://reactjs.org/docs/higher-order-components.html
Or we could extract this into an aux component, so we clean the `render()` of this component.

If you would like, we can explore a bit the magic of HOCs ✨